### ISSUE TITLE
Integrated converter into viewer link below

### ISSUE THREAD

**Issue by Isaina** [kovacsv/Online3DViewer#448]
https://github.com/RyugaRyuzaki/DotBimConvert

**Comment by GitHubDragonFly** [comment]
That DOTBimConvert appears to have small issues.

The exported [02.bim](https://github.com/RyugaRyuzaki/DotBimConvert/blob/main/Ifc/output/02/02.bim) file appears to have misplaced glass on the upper level tables.

The BIM converter that is included in the Online 3D Viewer appears to do a proper export of the original [02.ifc](https://github.com/RyugaRyuzaki/DotBimConvert/blob/main/Ifc/input/02.ifc) model.

**Comment by RyugaRyuzaki** [comment]
> That DOTBimConvert appears to have small issues.
> 
> The exported [02.bim](https://github.com/RyugaRyuzaki/DotBimConvert/blob/main/Ifc/output/02/02.bim) file appears to have misplaced glass on the upper level tables.
> 
> The BIM converter that is included in the Online 3D Viewer appears to do a proper export of the original [02.ifc](https://github.com/RyugaRyuzaki/DotBimConvert/blob/main/Ifc/input/02.ifc) model.

Yes, yes. issues, That why i need someone test with another file 
![image](https://github.com/kovacsv/Online3DViewer/assets/89787521/a986502a-65fc-4fb1-ba3a-c66ec7780604)

**Comment by Isaina** [comment]
Second thing is possible to write export plugin to ifc from .bim (ifc<->.bim), when converting from ifc to bim put meta data like Wall, Door, Window, and then using ifc.js so this Wall section become(IFCWall), Door become ifcDoor

 

Write IFC files with ifc.js ( node-typescript ):
https://www.youtube.com/watch?v=4EJyWjfPjB0

 

So we can convert IFC to .bim and back (maybe on first not all IFC elements covered just basic one ifcspace,ifcwall,ifcopenings,ifcwindos,ifcdoor)

 

This would be great

 

Best regards

 

Ivo 

 

 
 

Sent: Tuesday, February 13, 2024 at 11:25 AM
